lib/streamlit/config.py

```diff
@@ -749,6 +749,36 @@ def _browser_server_port() -> int:
     return int(get_option("server.port"))
 
 
+_SSL_PRODUCTION_WARNING = [
+    "DO NOT USE THIS OPTION IN A PRODUCTION ENVIRONMENT. It has not gone through "
+    "security audits or performance tests. For the production environment, "
+    "we recommend performing SSL termination by the load balancer or the reverse proxy."
+]
+
+_create_option(
+    "server.sslCertFile",
+    description=(
+        f"""
+        Server certificate file for connecting via HTTPS.
```

Comment:
@snehankekre These descriptions will be in the user-facing documentation. Do you want to look at it?

Reply:
These LGTM!